DCMAW-11008: Add test harness documentation (#408)
govuk-one-login/mobile-id-check-async: backend-api/docs/updatingTxmaTestHarness.md

Reviewer: On reflection I think this should live in the test-resources directory, perhaps in a docs folder.

Author: Moved to test-resources folder.

# How to update TxMA test harness when new events are being developed

> This guide provides the steps for updating the async test harness with a new
> TxMA event to ensure it is added to the correct files, and is available for
> testing.

Reviewer: ```suggestion
This guide provides the steps for updating the async test harness to process new TxMA events being developed in the async backend.
```

> ## Process

Reviewer: nit[take it or leave it] I think this is potentially a bit redundant

Author: Removed

### 1. Update the allowed events in the Dequeue Lambda

Update the [`getEvents`](../../test-resources/src/functions/dequeue/getEvent.ts)

> file to include the new TxMA event name.

Reviewer: nit:

Can this be something along the lines of 'Update the `allowedTxmaEventNames` array in `getEvents`(link) to include the new TxMA event being developed.

Author: Updated

###### Example

```typescript
export const allowedTxmaEventNames = [
  "DCMAW_ASYNC_CLIENT_CREDENTIALS_TOKEN_ISSUED",
  "DCMAW_ASYNC_CRI_START",
  "DCMAW_ASYNC_CRI_4XXERROR",
  "DCMAW_ASYNC_CRI_5XXERROR",
  "DCMAW_ASYNC_BIOMETRIC_TOKEN_ISSUED",
  "DCMAW_ASYNC_NEW_TXMA_EVENT_NAME",
];
```

### 2. Test the event is being dequeued correctly

> 1. Write the event to SQS by triggering the Lambda that sends the event.

Reviewer: ```suggestion
1. Write the event to the TxMA SQS by triggering the Lambda that sends the event. Alternatively, directly add a message to the TxMA SQS.
```
